src/net.h

```diff
@@ -416,6 +422,22 @@ class CConnman
     std::atomic<NodeId> nLastNodeId{0};
     unsigned int nPrevNodeCount{0};
 
+    // Cache responses to addr requests to minimize privacy leak.
+    // Attack example: scraping addrs in real-time may allow an attacker
+    // to infer new connections of the victim by detecting new records
+    // with fresh timestamps (per self-announcement).
+    struct CachedAddrResponse {
+        std::vector<CAddress> m_addrs_response_cache;
+        std::chrono::microseconds m_update_addr_response{0};
+    };
+
+    // Addr responses stored in different caches
+    // per network prevent cross-network node identification.
+    // If a node for example is multi-homed under Tor and IPv6,
+    // a single cache (or no cache at all) would let an attacker
+    // to easily detect that it is the same node by comparing responses.
+    std::map<Network, CachedAddrResponse> m_addr_response_caches;
```

Comment:
What are the additional memory requirements for this structure? Could you add that to the comment?